Marlin/src/HAL/STM32/tft/xpt2046.cpp

```diff
@@ -33,6 +33,7 @@
 uint16_t delta(uint16_t a, uint16_t b) { return a > b ? a - b : b - a; }
 
 SPI_HandleTypeDef XPT2046::SPIx;
+millis_t XPT2046::lastTouch;
```

Comment:
I suggest that all these added things be wrapped in `#if TOUCH_IDLE_SLEEP > 0` so they are never compiled unless the option is enabled.

Reply:
yep, i will try to move code from xpt class to Marlin/src/lcd/tft/touch.cpp, idle() func already check time... but its only for COLOR UI... so in TouchButtons class too (its one level higher but should work for all touchscreens then...)